app/Services/AdventuresLeagueAdapter.php

```diff
@@ -83,33 +84,80 @@ public function getCharacter($filePath): ?Character
+            $isDmLogEntry = ($data[$i][0] == "DmLogEntry");
+            $isTradeLogEntry = ($data[$i][0] == "TradeLogEntry");
+            $isPurchaseLogEntry = ($data[$i][0] == "PurchaseLogEntry");
+
+            $isEntryLine = ($isCharacterLogEntry || $isDmLogEntry || $isCampaignLogEntry || $isTradeLogEntry || $isPurchaseLogEntry);
 
-            if (!$isItemEntry) {
+            if ($isEntryLine) {
+                $type = "";
+                if ($isDmLogEntry) {
+                    $type = Entry::TYPE_DM;
+                } elseif ($isCampaignLogEntry || $isCharacterLogEntry) {
+                    $type = Entry::TYPE_GAME;
+                } else {
+                    $type = Entry::TYPE_DOWNTIME;
+                }
                 $entryData = [
                     'user_id' => Auth::id(),
+                    'adventure_id' => null,
+                    'campaign_id' => null,
                     'character_id' => $characterId,
-                    'date_played' => (array_key_exists(3, $data[$i])) ? $data[$i][3] : now(),
-                    'type' => Entry::TYPE_GAME,
+                    'event_id' => null,
+                    'dungeon_master_id' => null,
+                    'dungeon_master' => (array_key_exists(12, $data[$i])) ? (float)$data[$i][12] : "",
+                    'location' => (array_key_exists(11, $data[$i])) ? (float)$data[$i][11] : "",
+                    'levels' => 0,
+                    'date_played' => ($data[$i][3] == "") ? now() : $data[$i][3],
                     'gp' => (array_key_exists(7, $data[$i])) ? (float)$data[$i][7] : 0,
+                    'downtime' => (array_key_exists(8, $data[$i])) ? (float)$data[$i][8] : 0,
+                    'type' => $type,
                 ];
-                array_push($entries, new Entry($entryData));
-            }
-            // Item importing to be refactored
-            /*else {
-                //create the item and attach it to the last entry
-                $latestEntry = end($entries);
-                if (is_null($latestEntry)) {
-                    continue;
+                $entry = Entry::create($entryData);
+                $entry->save();
+                array_push($entries, $entry);
+            } elseif ($isItemEntry) {
+                if (count($entries) == 0) {
+                    $e = Entry::create([
+                        'user_id' => Auth::id(),
+                        'adventure_id' => null,
+                        'campaign_id' => null,
+                        'character_id' => $characterId,
+                        'event_id' => null,
+                        'dungeon_master_id' => null,
+                        'levels' => 0,
+                        'date_played' => now(),
+                        'gp' => 0,
+                        'downtime' => 0,
+                        'type' => Entry::TYPE_GAME,
+                    ]);
+                    $e->save();
+                    array_push($entries, $e);
                 }
```

Comment:
Ahhh i see whats happening here now, if theres no recently created entry we make one to attach the item to! We might want to consider extracting some of these control flow pieces to separate functions during our next refactor of this. Either way looks good for now :)

Reply:
yea this is specifically to handle the case where the first line is an item with no entry defined above it.